### DISCUSSION TITLE
Prisma error

### DISCUSSION THREAD

**Discussion by damiankomonski** [prisma/prisma#22381] [Q&A]

**Comment by damiankomonski** [comment]
Anyone could help?

**Reply by damiankomonski** [reply]
I dont know if i do this in good way.

**Comment by ludralph** [comment]
Hi @damiankomonski 👋 

Thank you for raising this question.

Your approach of checking each filter field and conditionally adding clauses to the where object is a common pattern when dealing with optional query parameters. You should note that if you have multiple AND arrays, you might need to merge them into a single AND array, as Prisma expects AND to be an array of conditions.

```js
// Colors
if (queryParameters.colors) {
    let colorsArray = queryParameters.colors.split(",");

    let orConditions = colorsArray.map(color => ({
        colors: {
            is: {
                name: color
            }
        }
    }));

    if (colorsArray.length === 1) {
        whereObject.colors = orConditions[0].colors;
    } else if (colorsArray.length > 1) {
        whereObject.AND = (whereObject.AND || []).concat([{ OR: orConditions }]);
    }
}
```

Can you try this revised code and see if it works for you?

**Reply by damiankomonski** [reply]
Hmm ok, so when I check every filter and add something new to object (which I create as where) is 100% valid? I have even long code.

**Reply by ludralph** [reply]
Yes, checking each filter and conditionally adding to the where object is a valid approach. This method allows you to dynamically build up your query based on the presence of optional filter parameters. The length of the code is not necessarily an issue as long as it is clear, maintainable, and performs the required filtering logic correctly.

**Comment by damiankomonski** [comment]
@ludralph Thank you very much for help.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
